### PR TITLE
Remove deprecated error codes.

### PR DESCRIPTION
**Pull request by TRodziewicz**
Signed-off-by: TRodziewicz <tomasz.rodziewicz@mobica.com>

## Description
Remove deprecated error codes, which can be found using `git grep -i 'MBEDTLS_ERR.*deprecated'.

Fixes: #4283 

## Status
IN DEVELOPMENT

## Requires Backporting
NO  

## Todos
- [x] Tests
- [x] Documentation
- [x] Changelog updated

### CODE REVIEW

**Comment by TRodziewicz** [comment]
Question about this task has been asked in the originating issue: https://github.com/ARMmbed/mbedtls/issues/4283#issuecomment-818232901

File: library/error.c [hunk 1]
```diff
@@ -110,6 +102,10 @@
 #include "mbedtls/error.h"
 #endif
 
+#if defined(MBEDTLS_PLATFORM_C)
+#include "mbedtls/platform.h"
```
Comment (mstarzyk-mobica) [inline library/error.c:106]: Same piece of code was removed at lines 165~168, is this intentional?

File: include/mbedtls/psa_util.h [hunk 2]
```diff
@@ -32,6 +32,10 @@
 #include MBEDTLS_CONFIG_FILE
 #endif
 
+#if defined(MBEDTLS_PLATFORM_C)
```
Comment (mstarzyk-mobica) [inline include/mbedtls/psa_util.h:35]: why is it needed here?

File: library/error.c [hunk 3]
```diff
@@ -740,11 +672,16 @@ const char * mbedtls_low_level_strerr( int error_code )
             return( "ERROR - This is a bug in the library" );
 #endif /* MBEDTLS_ERROR_C */
 
+#if defined(MBEDTLS_PLATFORM_C)
+        case -(MBEDTLS_ERR_PLATFORM_HW_ACCEL_FAILED):
```
Comment (mstarzyk-mobica) [inline library/error.c:676]: Since these error codes are now in error.h, do we still need the `#if defined(MBEDTLS_PLATFORM_C)` ?

**Review by mstarzyk-mobica — CHANGES_REQUESTED** [review]

File: ChangeLog.d/issue4283.txt [hunk 4]
```diff
@@ -0,0 +1,2 @@
+Removals
+   * Removed deprecated error codes. Fixes #4283
```
[the remark this answers is not in the document]
Comment (TRodziewicz) [reply ChangeLog.d/issue4283.txt]: Done

File: library/error.c (hunk 1 above)
Comment (TRodziewicz) [reply library/error.c:106]: Most probably yes, as the error.c is auto-generated by script.

File: include/mbedtls/psa_util.h (hunk 2 above)
Comment (TRodziewicz) [reply include/mbedtls/psa_util.h:35]: No! Removing...

File: library/error.c (hunk 3 above)
Comment (TRodziewicz) [reply library/error.c:676]: Good point, but as they were added by the script most probably the #ifdef block will be removed after the change above.

Comment (TRodziewicz) [reply library/error.c:676]: Seems that the either the `<generate_errors.pl>` or the `<error code names>` need to be modified to resolve this issue. I'm gonna ask the team.

File: library/error.c (hunk 1 above)
Comment (mstarzyk-mobica) [reply library/error.c:106]: oh, alright :)

File: library/error.c (hunk 3 above)
Comment (mstarzyk-mobica) [reply library/error.c:676]: I was not aware that error.c is autogenerated. It's fine with me to leave it as it is.

**Review by mstarzyk-mobica — DISMISSED** [review]

**Review by mpg — DISMISSED** [review]
This looks good to me, however there are new conflicts. We you fix them by merging `development_3.0` into your branch again? Then ping Mateusz and I, so that we re-review quickly and merge before other conflicts appear.

**Review by mpg — APPROVED** [review]
LGTM

**Review by mstarzyk-mobica — APPROVED** [review]

**Pull request merged**